Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -1814,29 +1823,20 @@ checkCabalVersion pkg =
 --
 checkPackageVersions :: GenericPackageDescription -> [PackageCheck]
 checkPackageVersions pkg =
-  catMaybes [
-
-    -- Check that the version of base is bounded above.
-    -- For example this bans "build-depends: base >= 3".
-    -- It should probably be "build-depends: base >= 3 && < 4"
-    -- which is the same as  "build-depends: base == 3.*"
-    check (not (hasUpperBound baseDependency)) $
-      PackageDistInexcusable BaseNoUpperBounds
-
-  ]
+  if length others > 0
```

Comment:
This is minor but instead of `if` is there a reason why you cannot do something like: `map (packageDistSuspiciousWarn . MissingUpperBounds) other : …`?

Reply:
Yes! That works, thanks!